src/server/scripts/World/npcs_special.cpp

```diff
@@ -2287,6 +2287,82 @@ struct npc_gen_void_zone : public ScriptedAI
     TaskScheduler _scheduler;
 };
 
+struct npc_bg_spirit_guide : public ScriptedAI
+{
+    static constexpr uint32 SPELL_SPIRIT_HEAL_CHANNEL = 22011;
+    static constexpr uint32 SPELL_GRAVEYARD_TELEPORT = 46893;
+
+    npc_bg_spirit_guide(Creature * creature) : ScriptedAI(creature) { }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+    void JustAppeared() override
+    {
+        ScheduleSpiritHealChannel();
+    }
+
+    void OnChannelFinished(SpellInfo const* /*spell*/) override
+    {
+        ScheduleSpiritHealChannel();
+    }
+
+    void OnDespawn() override
+    {
+        // this is a hack.
+        // A NPC is summoned (Alliance/Horde Graveyard Teleporter) and this NPC casts the spell.
+        DoCastSelf(SPELL_GRAVEYARD_TELEPORT);
```

Comment:
Any luck finding out what NPC is being summoned or if it's a spell summoning it? It might be serverside, but we have serverside spellIds up to Legion. If the creature is summoned, it should come up in the sniff with who created it and what spell created it.

Reply:
Possibly serverside spells:
- ID - 24237 Summon Alliance Graveyard Teleporter (SERVERSIDE) > summons 26350
- ID - 46894 Summon Horde Graveyard Teleporter (SERVERSIDE) > summons 26351

would need to figure out summon properties, need to set demon creator, needs to despawn as soon as possible (despawns at same time as spirit guide iirc) and make that npc cast the spell instead.
could also just summon the npc here without the spell and make it cast to be a bit more blizzlike

Reply:
I updated it to summon it already, but not with spells. Added comment with it